test/YAMLPlanExecutor.go

```diff
@@ -71,10 +88,90 @@ func ExecutePlanBuilder(path string, planBuilder *yaml.PlanBuilder) ([]byte, err
 			panic(fileErr)
 		}
 	}()
+
 	args := []string{"--plan"}
+	return executeShell(path, file, args...)
+}
+
+//ExecutePlanBuilderForApplication ...
+func ExecutePlanBuilderForApplication(path string, planBuilder *yaml.PlanBuilder, configuration config.Configuration) (statistics.AggregatorSnapShot, error) {
```

Comment:
This method is almost identical to `ExecutePlanFromDataForApplication` and `ExecuteListForApplication` so lets just do the writing to file and then call a generic method in all three of them.